src/main/scala/ai/privado/passes/DBTParserPass.scala

```diff
@@ -0,0 +1,462 @@
+
+    dbSchema
+  }
+
+  private def createDatabaseSinkRule(
+    projectName: String,
+    dbName: String,
+    dbPlatform: String,
+    dbHost: String,
+    schema: DatabaseSchema
+  ) = {
+    val ruleId = f"Storages.DBT.ReadAndWrite.${projectName}"
+
+    val customDatabaseSinkRule = RuleInfo(
+      ruleId,
+      f"${dbName}",
+      "",
+      Array[String]("getdbt.com"),
+      List[String](),
+      false,
+      "",
+      HashMap[String, String](),
+      NodeType.REGULAR,
+      "",
+      CatLevelOne.SINKS,
+      "storages",
+      Language.DEFAULT,
+      Array[String]()
+    )
+
+    val dbDetails = DatabaseDetails(dbName, dbPlatform, dbHost, "", "", Some(schema))
+
+    ruleCache.setRuleInfo(customDatabaseSinkRule)
+    DatabaseDetailsCache.addDatabaseDetails(dbDetails, ruleId)
+
+    customDatabaseSinkRule
+  }
+
+  private def createSQLNode(builder: DiffGraphBuilder, dbNode: NewDbNode) = {
+    val sqlNode = NewSqlQueryNode().name("DBT").code("DBT").order(0)
+    builder.addNode(sqlNode)
+    builder.addEdge(dbNode, sqlNode, EdgeTypes.AST)
+
+    sqlNode
+  }
+  private def createAndTagDatabaseNode(
+    builder: DiffGraphBuilder,
+    ruleInfo: RuleInfo,
+    projectFile: String,
+    projectName: String,
+    profileFile: String,
+    dbName: String,
+    dbKey: String
+  ) = {
+    val (fileName, (lineNumber, columnNumber, matchedLine)) = dbName match {
+      case "DBT" => (projectFile, findFirstPatternInYAMLFile(projectFile, "name", projectName))
+      case _     => (profileFile, findFirstPatternInYAMLFile(profileFile, dbKey, dbName))
+    }
+
+    val fileNode = addFileNode(fileName, builder)
+
+    val databaseNode = NewDbNode()
+      .name(dbName)
+      .code(matchedLine)
+      .lineNumber(lineNumber)
+      .columnNumber(columnNumber)
+      .order(0)
+
+    builder.addNode(databaseNode)
+    builder.addEdge(databaseNode, fileNode, EdgeTypes.SOURCE_FILE)
+
+//    Utilities.addRuleTags(builder, databaseNode, ruleInfo, ruleCache)
+
+    val storeForTagHelper = storeForTag(builder, databaseNode) _
+    storeForTagHelper(Constants.id, ruleInfo.id)
+    storeForTagHelper(Constants.nodeType, ruleInfo.nodeType.toString)
+    storeForTagHelper(Constants.catLevelOne, ruleInfo.catLevelOne.name)
+    storeForTagHelper(Constants.catLevelTwo, ruleInfo.catLevelTwo)
+
+    databaseNode
+  }
+
+  private def storeForTag(
+    builder: BatchedUpdate.DiffGraphBuilder,
+    node: AstNodeNew
+  )(tagName: String, tagValue: String = ""): BatchedUpdate.DiffGraphBuilder = {
+    builder.addEdge(node, NewTag().name(tagName).value(tagValue), EdgeTypes.TAGGED_BY)
+    builder
+  }
+
+  private def createTableColumnNodesFromModels(
+    builder: DiffGraphBuilder,
+    model: java.util.Map[String, Any],
+    tableQueryOrder: Int
+  ) = {
+    val modelFile = model.get("filePath").asInstanceOf[String]
+    val tableName = model.get("name").asInstanceOf[String]
```

Comment:
Statements like these can cause runtime exception, handle such statements

Reply:
In this case, these values are already confirmed in the previous function.